<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="two_d_mode" default="true" description="If true, calibration will only be done in 2 dimensions instead of 3." />
  <arg name="calibration_file_path" default="$(env HOME)/.ros/magnetometer_calib.yaml" description="If nonempty, specifies path to the calibration file." />
  <arg name="load_from_params" default="true" description="Whether to load initial calibration from ROS params." />
  <arg name="load_from_file" default="false" description="Whether to load initial calibration from calibration file." />
  <arg name="save_to_file" default="false" description="Whether to save updated calibration to calibration file." />
  <arg name="initial_mag_bias_x" default="0.0" description="BAG FILE." />
  <arg name="initial_mag_bias_y" default="0.0" description="BAG FILE." />
  <arg name="initial_mag_bias_z" default="0.0" description="BAG FILE." />
  
  <!-- PY bias observer node -->
  <node pkg="magnetometer_pipeline" exec="magnetometer_bias_observer.py" name="mag_bias_observer" respawn="true">
    <!-- <remap from="imu/mag" to="magnetometer/data" /> -->
    <param name="2d_mode" value="$(var two_d_mode)" />
    <param name="calibration_file_path" value="$(var calibration_file_path)" />
    <param name="load_from_params" value="$(var load_from_params)" />
    <param name="load_from_file" value="$(var load_from_file)" />
    <param name="save_to_file" value="$(var save_to_file)" />
  </node>
  
  <!-- CPP bias remover node -->
  <node pkg="magnetometer_pipeline" exec="magnetometer_bias_remover_nodelet" name="magnetometer_bias_remover_nodelet">
    <!-- <remap from="imu/mag" to="magnetometer/data" /> -->
    <param name="initial_mag_bias_x" value="$(var initial_mag_bias_x)" />
    <param name="initial_mag_bias_y" value="$(var initial_mag_bias_y)" />
    <param name="initial_mag_bias_z" value="$(var initial_mag_bias_z)" />
  </node>
</launch>